### PR TITLE
Wu et al. 2021

### LINKED ISSUE #94: Wu et al 2021
## Paper

Wu R, Li H, Li R, Peng D, Wang N, Shen X, Sun H (2021) Identification and sequencing of 59 highly polymorphic microhaplotypes for analysis of DNA mixtures. *International Journal of Legal Medicine*, 135:1137-1149, [doi:10.1007/s00414-020-02483-x](https://doi.org/10.1007/s00414-020-02483-x).

## Data URL(s)

Table S2

[Wu etal suppl data_ESM.xlsx](https://github.com/bioforensics/MicroHapDB/files/9891467/Wu.etal.suppl.data_ESM.xlsx)

[Wu etal suppl data_ESM.txt](https://github.com/bioforensics/MicroHapDB/files/9891475/Wu.etal.suppl.data_ESM.txt)

https://doi.org/10.1007/s00414-020-02483-x

## Author and PI information

Hongyu Sun, sunhy@mail.sysu.edu.cn

### PR DESCRIPTION
**Pull request by standage**
This branch adds a new marker collection from [Wu et al. (2021)](https://doi.org/10.1007/s00414-020-02483-x). Closes #94.

## Effective number of alleles (A<sub>e</sub>)

avg_ae_26_vs_3

This plot shows A<sub>e</sub> values averaged over the three Chinese population groups in the 1000 Genomes Project as a function of A<sub>e</sub> averaged over all 26 global population groups. Glyphs above the dashed grey line at `y=x` indicate markers with more allelic diversity in Chinese populations than the global average. Markers from this study are shown in red, with markers from USC and ALFRED shown in green and blue for comparison.

avg_ae_26_vs_chinese

This plot shows A<sub>e</sub> values in three distinct Chinese population groups as a function of the global average A<sub>e</sub>.

### CODE REVIEW

**Comment by standage** [comment]
Data files:

- [KK/ALFRED kkk_aes.txt](https://github.com/bioforensics/MicroHapDB/files/9912528/kkk_aes.txt)
- [Wu 2021 shy_aes.txt](https://github.com/bioforensics/MicroHapDB/files/9912529/shy_aes.txt)
- [de la Puente 2020 usc_aes.txt](https://github.com/bioforensics/MicroHapDB/files/9912530/usc_aes.txt)

File: dbbuild/marker.tsv [hunk 1]
```diff
@@ -25,7 +26,7 @@ mh01USC-1pD	MHDBM-8098d345	GRCh38	chr1	106227453,106227475,106227487	3.1574	0.37
 mh01KK-211	MHDBM-6adc547b	GRCh38	chr1	161985865,161985967,161986013	2.2742	0.1706	0.1164	ALFRED
 mh01AT-03	MHDBM-be82bac2	GRCh38	chr1	161985967,161985984,161986013	2.7654	0.1143	0.0071	ISFG2019:P597
 mh01USC-1qA	MHDBM-f218f0ad	GRCh38	chr1	167157730,167157749	2.5003	0.0976	0.0695	10.1016/j.fsigen.2019.102213
-mh01NH-03	MHDBM-e7a95c5e	GRCh38	chr1	184807944,184807966,184808042	1.9085	0.1987	0.1725	10.1016/j.legalmed.2015.06.003
+mh01NH-03	MHDBM-e7a95c5e	GRCh38	chr1	184807944,184807966,184808042	1.9086	0.1987	0.1725	10.1016/j.legalmed.2015.06.003
```
Comment (standage) [inline dbbuild/marker.tsv:29]: A couple of these rounding artifacts pop up every once in a while depending on numpy version. 🤨

File: dbbuild/sources/1kgp/mhfreqs.py [hunk 2]
```diff
@@ -46,6 +46,9 @@ def construct_haplotypes(rsids, vcf, rsidxfile):
                     samples = line.strip().split()[9:]
                 continue
             fields = line.strip().split()
+            rsid = fields[2]
+            if rsid not in rsids:
+                continue
```
Comment (standage) [inline dbbuild/sources/1kgp/mhfreqs.py:51]: One of the markers is spanned by two CNVs, resulting in bookkeeping errors when aggregating the marker SNP data. These change to the build code eliminates that error.

**Comment by rnmitchell** [comment]
Will do this today!

**Review by rnmitchell — APPROVED** [review]
Looks good!

**Comment by standage** [comment]
Thanks @rnmitchell!

**Pull request merged**